client/api.c

```diff
@@ -704,6 +707,49 @@ TDNFOpenHandle(
                   TDNF_CONF_GROUP);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    if (!IsNullOrEmptyString(pTdnf->pArgs->pszInstallRoot) &&
+        strcmp(pTdnf->pArgs->pszInstallRoot, "/"))
+    {
+        int nIsDir = 0;
+
+        dwError = TDNFAllocateStringPrintf(&pszCacheDir, "%s/%s",
+                                           pTdnf->pArgs->pszInstallRoot,
+                                           pTdnf->pConf->pszCacheDir);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        TDNF_SAFE_FREE_MEMORY(pTdnf->pConf->pszCacheDir);
+        pTdnf->pConf->pszCacheDir = pszCacheDir;
+        pszCacheDir = NULL;
```

Comment:
Why is it assigned to NULL here and in cleanup there is call to TDNF_SAFE_FREE_MEMORY(pszCacheDir)?  Is it done on purpose?

Reply:
pszCacheDir is re-used in this function and the pointer is stored in  pTdnf->pConf->pszCacheDir and later this is  freed in TDNFCloseHandle?

Reply:
My point was there is no use of calling TDNF_SAFE_FREE_MEMORY(pszCacheDir) if pszCacheDir is assigned to NULL since TDNF_SAFE_FREE_MEMORY doesnt do anything is pointer is NULL. Also, TDNFCloseHandle calls TDNFFreeConfig which frees TDNF_SAFE_FREE_MEMORY(pConf->pszCacheDir). Wouldn't this cause a leak to pszCacheDir pointer?

Reply:
Nevermind, there is no leak here as the pointer is freed in TDNFFreeConfig, but there is no use of calling TDNF_SAFE_FREE_MEMORY(pszCacheDir) since pszCacheDir is assigned to NULL.